markdownlint.sh

```diff
@@ -1,64 +1,113 @@
-branch=$( cat $ATOMIST_PAYLOAD |
-  jq -r '.data.Push[0].branch' )
+declare Pkg=markdownlint
+declare Version=0.1.0
 
-##### Bail out early if it on a markdownlint branch
-if [[ "$branch" =~ ^markdownlint-.* ]]; then
-    exit 0
-fi
+set -o pipefail
 
-##### Make the problem matcher available to the runtime
-cp /app/markdownlint.matcher.json /atm/output/matchers/
+# print message to stdout prefixed by package name.
+# usage: msg MESSAGE
+function msg () {
+    echo "$Pkg: $*"
+}
 
-##### Create push instructions for the runtime to indicate how changes to the repo should get persisted
-if [[ -z "$labels" ]]; then
-  labels="[]"
-fi
-push=$( jq -n \
-    --arg s "$fix" \
-    --argjson l "$labels" \
-    '{
-        strategy: $s,
-        pullRequest: {
-          title: "MarkdownLint fixes",
-          body: "MarkdownLint fixed warnings and/or errors",
-          branchPrefix: "markdownlint",
-          labels: $l,
-          close: {
-            stale: true,
-            message: "Closing pull request because all fixable warnings and/or errors have been fixed in base branch"
-          }
-        },
-        commit: {
-          message: "MarkdownLint fixes"
-        }
-    }'
-    )
-echo $push > /atm/output/push.json
+# print message to stderr prefixed by package name.
+# usage: err MESSAGE
+function err () {
+    msg "$*" 1>&2
+}
 
-##### Prepare command arguments
-if ! [[ -z "$fix" ]]; then
-  fix_option="--fix"
-fi
+function main () {
+    # Extract some skill configuration from the incoming event payload
+    local fix config ignores labels branch
+    fix=$( < "$ATOMIST_PAYLOAD" \
+               jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "push_strategy" ) | .value' )
+    config=$( < "$ATOMIST_PAYLOAD" \
+                  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "config" ) | .value' )
+    ignores=$( < "$ATOMIST_PAYLOAD" \
+                   jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "ignores" ) | .value | join(" --ignore ")' )
+    labels=$( < "$ATOMIST_PAYLOAD" \
+                  jq -r '.skill.configuration.instances[0].parameters[] | select( .name == "labels" ) | .value' )
+    branch=$( < "$ATOMIST_PAYLOAD" jq -r '.data.Push[0].branch' )
 
-if ! [[ -z "$config" ]] && ! [[ -f "/atm/home/.markdownlint.json" ]]; then
-  echo $config > "/atm/input/markdownlint.config.json"
-  config_option="--config /atm/input/markdownlint.config.json"
-fi
+    # Bail out early if it on a markdownlint branch
+    if [[ $branch == markdownlint-* ]]; then
+        exit 0
+    fi
 
-if ! [[ -z "ignores" ]]; then
-  ignore_option="--ignore $ignores"
-fi
+    local outdir=${ATOMIST_OUTPUT_DIR:-/atm/output}
 
-markdownlint "**/*.md" $config_option $ignore_option $fix_option
+    # Make the problem matcher available to the runtime
+    local matcher_outdir=$outdir/matchers
+    if ! mkdir -p "$matcher_outdir"; then
```

Comment:
That directory will always exist. The runtime creates it. Is that going to cause an issue?

Reply:
```man
     -p      Create intermediate directories as required.  If this option is
             not specified, the full path prefix of each operand must already
             exist.  On the other hand, with this option specified, no error
             will be reported if a directory given as an operand already
             exists.  Intermediate directories are created with permission
             bits of rwxrwxrwx (0777) as modified by the current umask, plus
             write and search permission for the owner.
```

If it always exists, is it provided as an environment variable or just a convention?